### PR TITLE
feat(query): type decimal support agg func min/max

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

type decimal support agg func min/max

Closes #part of #10005

### CODE REVIEW

File: src/query/expression/src/types/decimal.rs [hunk 1]
```diff
@@ -117,6 +245,53 @@ impl Decimal for i128 {
...
+
+    fn try_downcast_domain(domain: &DecimalDomain) -> Option<SimpleDomain<Self>> {
+        match domain {
+            DecimalDomain::Decimal128(val, _) => Some(*val),
+            _ => None,
+        }
+    }
+
+    // will mock DecimalSize need modify when use it
+    fn upcast_scalar(scalar: Self) -> Scalar {
```
Comment (TCeason) [inline src/query/expression/src/types/decimal.rs:271]: Now will mock a size.

File: tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal [hunk 2]
```diff
@@ -112,3 +112,18 @@ query I
 SELECT CAST(-4.56 AS DECIMAL(6, 2)) / CAST(-1.23 AS DECIMAL(6, 2)) AS result;
 ----
 3.69
+
+query I
+SELECT MIN(CAST(-4.56 AS DECIMAL(6, 2)))
```
Comment (sundy-li) [inline tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal:117]: Need tests that scan more than one row.

Comment (TCeason) [reply tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal:117]: I try to create select union select to replace create table but failed.

Comment (sundy-li) [reply tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal:117]: such as `select max(number::Decimal(6,2)) from numbers(10)`

**Pull request merged**